File: Projekt2_62280.py
class Car:
    def __init__(self, brand, model, car_type, price_per_day):
        self.brand = brand
        self.model = model
        self.car_type = car_type
        self.price_per_day = price_per_day

    def __str__(self):
        return f"{self.brand} {self.model} ({self.car_type}) - {self.price_per_day} zł/dzień"


class Customer:
    def __init__(self, name):
        self.name = name

    def __str__(self):
        return f"{self.name}"


class Rental:
    def __init__(self, customer, car, days):
        self.customer = customer
        self.car = car
        self.days = days

    def calculate_cost(self):
        return self.days * self.car.price_per_day

    def reservation_summary(self):
        return (f"\n--- Podsumowanie Rezerwacji ---\n"
                f"Klient: {self.customer}\n"
                f"Samochód: {self.car}\n"
                f"Liczba dni: {self.days}\n"
                f"Koszt całkowity: {self.calculate_cost()} zł\n")


def show_menu(cars):
    print("=== Dostępne samochody ===")
    for idx, car in enumerate(cars):
        print(f"{idx + 1}. {car}")


def main():
    cars = [
        Car("AUDI", "RS3 8Y", "SEDAN", 899),
        Car("Mercedes", "A200", "SEDAN", 299),
        Car("Mercedes ", "G63 AMG", "SUV", 2799),
        Car("BMW", "M240i", "COUPE", 799),
        Car("BMW", "X7", "SUV", 1299),
        Car("Volkswagen", "Golf 7R", "Hatchback", 549)
    ]

    name = input("Podaj imię i nazwisko klienta: ")
    customer = Customer(name)

    show_menu(cars)
    choice = int(input("Wybierz numer samochodu: ")) - 1

    if choice < 0 or choice >= len(cars):
        print("Nieprawidłowy wybór.")
        return

    days = int(input("Podaj liczbę dni wynajmu: "))
    rental = Rental(customer, cars[choice], days)

    print(rental.reservation_summary())


if __name__ == "__main__":
    main()

File: Projekt2_62280.cs
using System;


public class Car
{
    public string Brand { get; }
    public string Model { get; }
    public string CarType { get; }
    public decimal PricePerDay { get; }

    public Car(string brand, string model, string carType, decimal pricePerDay)
    {
        Brand = brand;
        Model = model;
        CarType = carType;
        PricePerDay = pricePerDay;
    }

    public override string ToString()
    {
        return $"{Brand} {Model} ({CarType}) - {PricePerDay} zł/dzień";
    }
}

public class Customer
{
    public string Name { get; }

    public Customer(string name)
    {
        Name = name;
    }

    public override string ToString()
    {
        return Name;
    }
}

public class Rental
{
    public Customer Customer { get; }
    public Car Car { get; }
    public int Days { get; }

    public Rental(Customer customer, Car car, int days)
    {
        Customer = customer;
        Car = car;
        Days = days;
    }

    public decimal CalculateCost()
    {
        return Days * Car.PricePerDay;
    }

    public void PrintSummary()
    {
        Console.WriteLine("\n--- Podsumowanie Rezerwacji ---");
        Console.WriteLine($"Klient: {Customer}");
        Console.WriteLine($"Samochód: {Car}");
        Console.WriteLine($"Liczba dni: {Days}");
        Console.WriteLine($"Koszt całkowity: {CalculateCost()} zł");
    }
}

class Program
{
    static void Main()
    {
        List<Car> cars = new List<Car>
        {
        new Car("AUDI", "RS3 8Y", "SEDAN", 899),
        new Car("Mercedes", "A200", "SEDAN", 299),
        new Car("Mercedes ", "G63 AMG", "SUV", 2799),
        new Car("BMW", "M240i", "COUPE", 799),
        new Car("BMW", "X7", "SUV", 1299),
        new Car("Volkswagen", "Golf 7R", "Hatchback", 549)
        };

        Console.Write("Podaj imię i nazwisko klienta: ");
        string name = Console.ReadLine();
        Customer customer = new Customer(name);

        Console.WriteLine("\n=== Dostępne samochody ===");
        for (int i = 0; i < cars.Count; i++)
        {
            Console.WriteLine($"{i + 1}. {cars[i]}");
        }

        Console.Write("Wybierz numer samochodu: ");
        int choice = int.Parse(Console.ReadLine()) - 1;

        if (choice < 0 || choice >= cars.Count)
        {
            Console.WriteLine("Nieprawidłowy wybór.");
            return;
        }

        Console.Write("Podaj liczbę dni wynajmu: ");
        int days = int.Parse(Console.ReadLine());

        Rental rental = new Rental(customer, cars[choice], days);
        rental.PrintSummary();
    }
}
